Default config tests

MKashevsky opened the pull request.

MKashevsky committed:
moved config to separate fixture
moved temp args from config constructor to fixture
added deepcopy for config.__copy__

MKashevsky committed:
updated Node class
set config as optional parameter
paths and chain_params set as class attributes
refactored setup method

MKashevsky committed:
set generation of config and genesis files optional

MKashevsky committed:
set dummy port if config is empty in Docker_controller.run_node
Config class extended with get method

MKashevsky committed:
added fixture containing default config from scorumd
user accounts moved to separated list
added fixtures for accounts and witness account
added read method to Config class
added environment param on container run in DockerController

MKashevsky committed:
added simple logs checks

IvanRomanovsky commented on tests/conftest.py:
```diff
@@ -8,21 +8,83 @@
+            return acc
+
+
+@pytest.fixture(scope='session')
```
remove this fixture

MKashevsky replied:
removed

IvanRomanovsky commented on tests/conftest.py:
```diff
@@ -8,21 +8,83 @@
+
+
+@pytest.fixture(scope='session')
+def witness(accounts):
```
remove this fixture

MKashevsky replied:
removed

IvanRomanovsky commented on tests/conftest.py:
```diff
@@ -8,21 +8,83 @@
     parser.addoption('--use-local-image', action='store_false', help='dont rebuild image')
 
 
+def create_acc(
```
hide create account logic into genesis module

MKashevsky replied:
done

IvanRomanovsky commented on tests/test_single_node.py:
```diff
@@ -19,22 +18,44 @@ def test_block_production(wallet: Wallet, node: Node):
 
     assert Amount(info['total_supply']) == expected_total_supply
-    assert wallet.get_account_scr_balance(initdelegate.name) == Amount('80.000000000 SCR')
+    assert wallet.get_account_scr_balance(creator) == Amount('80.000000000 SCR')
```
add balance check for all account in genesis

MKashevsky replied:
added

MKashevsky committed:
account creation logic moved to genesis class
supply functions and attributes removed from account class
updated default config fixture to wait for config generation
removed witness and creator fixtures
updated test_genessis_block to check all accounts
